transactions/transactor_test.go

```diff
@@ -385,41 +388,126 @@ func (s *TxQueueTestSuite) TestLocalNonce() {
 	s.Equal(uint64(nonce)+1, resultNonce.(uint64))
 }
 
-func (s *TxQueueTestSuite) TestContractCreation() {
-	key, _ := crypto.GenerateKey()
-	testaddr := crypto.PubkeyToAddress(key.PublicKey)
-	genesis := core.GenesisAlloc{
-		testaddr: {Balance: big.NewInt(100000000000)},
-	}
-	backend := backends.NewSimulatedBackend(genesis)
-	selectedAccount := &account.SelectedExtKey{
-		Address:    testaddr,
-		AccountKey: &keystore.Key{PrivateKey: key},
-	}
-	s.manager.sender = backend
-	s.manager.gasCalculator = backend
-	s.manager.pendingNonceProvider = backend
-	tx := SendTxArgs{
-		From:  testaddr,
-		Input: hexutil.Bytes(gethcommon.FromHex(contract.ENSBin)),
+func (s *TxQueueTestSuite) TestTransactionsWithSimulatedBackend() {
+
+	testCases := []struct {
+		name             string
+		getTransactions  func(testaddr common.Address) []SendTxArgs
+		receiptValidator func(receipt *types.Receipt, testaddr common.Address)
+	}{
+		{
+			"EthTransfer",
+			func(testaddr common.Address) []SendTxArgs {
+				return []SendTxArgs{
+					{
+						Value: (*hexutil.Big)(big.NewInt(10)),
+					},
+				}
+			},
+			func(receipt *types.Receipt, testaddr common.Address) {
+				successStatus := uint(1)
+				s.Equal(successStatus, receipt.Status)
+			},
+		},
+		{
+			"ContractCreation",
+			func(testaddr common.Address) []SendTxArgs {
+				return []SendTxArgs{
+					{
+						Input: hexutil.Bytes(gethcommon.FromHex(contract.ENSBin)),
+					},
+				}
+			},
+			func(receipt *types.Receipt, testaddr common.Address) {
+				s.Equal(crypto.CreateAddress(testaddr, 0), receipt.ContractAddress)
+			},
+		},
+		{
+			"ContractStateTransfer",
+			func(testaddr common.Address) []SendTxArgs {
+				// method ID: web3.utils.sha3('setOwner(bytes32,address)').substring(0, 8)
+				setOwnerMethodID := "0x5b0fc9"
+				node := make([]byte, 32)
+				ensParsed, _ := abi.JSON(strings.NewReader(contract.ENSABI))
+				setOwnerMethodData, _ := ensParsed.Pack("setOwner", setOwnerMethodID, node, testaddr)
+
+				// method ID: web3.utils.sha3('owner(bytes32)').substring(0, 8)
+				ownerMethodID := "0x02571b"
+				getOwnerMethodData, _ := ensParsed.Pack("owner", ownerMethodID, node)
+
+				return []SendTxArgs{
+					{
+						// Create contract
+						Input: hexutil.Bytes(gethcommon.FromHex(contract.ENSBin)),
+					},
+					{
+						// Change contract owner
+						Input: hexutil.Bytes(setOwnerMethodData),
+					},
+					{
+						// Check that the contract owner is correct
+						Input: hexutil.Bytes(getOwnerMethodData),
+					},
+				}
+			},
+			func(receipt *types.Receipt, testaddr common.Address) {
+				successStatus := uint(1)
```

Comment:
@dshulyak This PR should be ready. The only thing missing is getting the contract owner in the response to the last transaction in this last test case. Currently we are only checking that there is a success status in the transaction receipt, but I need to figure out how to get the response to the Owner method in the contract in order to double check that there was an owner change for that particular ENS node

Reply:
Still need to figure out how to do this, but it turns out the tests were passing because they were cached. Will fix them and take off the WIP tag again when its done